maxvolume permissions

KingGold171 opened the pull request:
Added - MaxVolume in permissions (Limit how loud people can set the bots
volume, the number set in the permissions.ini is the limit)

imayhaveborkedit committed:
Fuck with init stuff, again

Unclosed client session warnings are dumb, will fix properly eventually.

imayhaveborkedit committed:
Try to mitigate the restart loop madness

imayhaveborkedit committed:
Refactored Entry objects into their own file

imayhaveborkedit committed:
More function moving

imayhaveborkedit committed:
Deleted terrible unused function

imayhaveborkedit committed:
Added some optimistic ffmpeg options

These may or may not do anything

imayhaveborkedit committed:
Improved some invite url stuff

The bot now caches the app id and prints it when the bot isn't on any
servers at startup

imayhaveborkedit committed:
potato

imayhaveborkedit committed:
Entries can now be skipped by their queuer.

imayhaveborkedit committed:
minor tweak blah

imayhaveborkedit committed:
mfw "how do I get the OwnerID"

imayhaveborkedit committed:
meh

imayhaveborkedit committed:
More requirements tricks

Discord.py was also updated on pypi

imayhaveborkedit committed:
Bump version

imayhaveborkedit committed:
Redirect and filter stderr from the ffmpeg subproc

imayhaveborkedit committed:
Identify if joining Discord API

imayhaveborkedit committed:
Fix channel printing with invalid entries

imayhaveborkedit committed:
This had better fix this option

imayhaveborkedit committed:
Updated scripts

imayhaveborkedit committed:
Moved ffmpeg binaries to a folder

imayhaveborkedit committed:
Oh right redundant line

imayhaveborkedit committed:
Set executable bit

imayhaveborkedit committed:
Finish up server join mention stuff

imayhaveborkedit committed:
Not sure if I want these

imayhaveborkedit committed:
Fixed potential zerodivisionerror

imayhaveborkedit committed:
Message deletion option fixes

Invoking messages should be deleted even if there's an error sending the
message
Delete invoking messages even if no response is given

imayhaveborkedit committed:
Print this thing I guess

Might be None sometimes idk

imayhaveborkedit committed:
Implement error handling from ffmpeg stderr

imayhaveborkedit committed:
Unretardify future exception check

If for some reason that future wasn't done by that point it would send
back an InvalidStateError

imayhaveborkedit committed:
Add mostly complete streaming support

imayhaveborkedit committed:
alright change of plans

imayhaveborkedit committed:
#400

imayhaveborkedit committed:
Fix nazi autoplaylist item removing

imayhaveborkedit committed:
...and remove the old code

imayhaveborkedit committed:
Move _fixg to utils

imayhaveborkedit committed:
Remove unused function

imayhaveborkedit committed:
Move safe_print to utils

imayhaveborkedit committed:
Move patchedbuff _avg to utils

imayhaveborkedit committed:
Fixed not updating resolved stream url

imayhaveborkedit committed:
Update ffmpeg messages

imayhaveborkedit committed:
These are functions

imayhaveborkedit committed:
ffmpeg stderr thread exception handling fixes

imayhaveborkedit committed:
Add source_url to stream entries

The input url should be given, not the direct url
For example, the np command

imayhaveborkedit committed:
Bit of random cleanup

Good thing I never used the user functions

imayhaveborkedit committed:
Fix dumb NoneType printing

imayhaveborkedit committed:
Rename and update autoplaylist

This is basically just moving the autoplaylist file so it doesn't
conflict, like with the example_ files.
Also moved some code around in the config code and added the check for
the autoplaylist file

imayhaveborkedit committed:
Print bot tag on startup

imayhaveborkedit committed:
hurhur

imayhaveborkedit committed:
Add bins folder to path on windows

imayhaveborkedit committed:
Begone filthy resource warning

imayhaveborkedit committed:
Make sure new __del__ doesn't do anything dumb

imayhaveborkedit committed:
Added deletion to remove_from_autoplaylist

The function can now optionally delete the entry from the file

imayhaveborkedit committed:
Never again pollute my issue tracker with this shit

If I see this shit one more time I'm rewriting this entire system.

imayhaveborkedit committed:
wew lad lets continue on autoplaylist errors

imayhaveborkedit committed:
Updated appinfo stuff + decorator

imayhaveborkedit committed:
Use sysroot var in py.exe invoke

Stupid path issues

imayhaveborkedit committed:
A few fixes

imayhaveborkedit committed:
whoops safe_print

imayhaveborkedit committed:
Reworked some config code and added auto OwnerID

Also updated config.ini with better comments and limited line length

imayhaveborkedit committed:
Proofreading example config hurdur

imayhaveborkedit committed:
Forgot to close my session too

imayhaveborkedit committed:
Thanks jake :+1:

imayhaveborkedit committed:
Fix autoplaylist remove function and gitignore

imayhaveborkedit committed:
Moved classes to constructs file

Also wrote a VoiceStateUpdate class that will be used in a later commit

imayhaveborkedit committed:
Added ignore entry and comments

imayhaveborkedit committed:
Voice code rewrite + other stuff

Big bulky commit.  Lets hope that the voice connection problems stop
being problems.

- Moved from manual multivoice impl to library voice impl
- Fixed resume related logic
- Rewrote voice state update code
- Added/changed some messages and errors

imayhaveborkedit committed:
Stupid...

imayhaveborkedit committed:
comments...

imayhaveborkedit committed:
Fix format string derp

imayhaveborkedit committed:
Improve voice channel property

imayhaveborkedit committed:
Add discord return types to VoiceStateUpdate props

imayhaveborkedit committed:
Removed dumb property and changed empty default

imayhaveborkedit committed:
Finished up on_voice_state_update rewrite

imayhaveborkedit committed:
Consolidate autojoin/summon code

Autojoining and autosummoning is now all in one function.  The bot
should now autosummon if it finds the owner in a voice channel and
there's not an autojoin channel set for that server.

Additionally, the bot will now resume voice connections if possible.
This means if it sees itself in a voice channel it'll join that voice
channel again so it won't appear in it without being in it.

imayhaveborkedit committed:
Removed unneeded check

imayhaveborkedit committed:
Unify server/channel naming scheme

imayhaveborkedit committed:
I suppose this is what you would call a regression

Plus other voice state stuff

imayhaveborkedit committed:
Move Serializable to constructs.py

imayhaveborkedit committed:
Finish autopause stuff

This adds a check during startup and adds a check in the voice state
event

imayhaveborkedit committed:
Debug msg clarification and misc tweaks

imayhaveborkedit committed:
Moved some init variables around

imayhaveborkedit committed:
Add server join/part msgs and kill player on part

imayhaveborkedit committed:
Pause/resume players based on server availability

Also added messages for availability changes

imayhaveborkedit committed:
This didn't need to be async

imayhaveborkedit committed:
Icky code cleanup

imayhaveborkedit committed:
Add ready and debug mode checks

imayhaveborkedit committed:
Add dev commands

imayhaveborkedit committed:
Mess with imports

imayhaveborkedit committed:
Fix reconnect function and websocket check

The voice update event was interfering with the reconnect code so I just
needed to add locks.  Worst case scenario I missed a deadlock condition,
second worst case scenario the bot might reconnect twice.

imayhaveborkedit committed:
Misc tweaks and changes

Fixed bad owner check
Added placeholder httpexception check condition
Tweaked help command text a bit
Breakpoints are weird sometimes

imayhaveborkedit committed:
Bump version

Don't think we're done yet, oh no.

imayhaveborkedit committed:
Small change to reconnect_voice_client

This makes a bit more sense.  I had a weird issue with reconnecting in a
infinite loop.

imayhaveborkedit committed:
Voice code stuff

lets see if we can fix this one last issue

imayhaveborkedit committed:
More connection debug code

imayhaveborkedit committed:
Initial logging implementation

I'm just committing what I've converted to logging for now

imayhaveborkedit committed:
Added more logging code and misc changes

Changed some formatters from old to new
Changed some :emoji: to \N{X} format
Random tweaks

imayhaveborkedit committed:
More logging stuff

imayhaveborkedit committed:
Started sanity checks

imayhaveborkedit committed:
Finished converting most of the code to logging...

and a few misc changes/tweaks/whatever

imayhaveborkedit committed:
Removed unneeded class

imayhaveborkedit committed:
Implemented color logging and new log levels

Added colorlog to requirements.txt
Switched console logging to colorlog logger
Added VOICEDEBUG, NOISY and EVERYTHING logging levels
Changes some logging settings

imayhaveborkedit committed:
Fixed format error

imayhaveborkedit committed:
Add more VoiceStateUpdate functionality

Also add slots to VoiceStateUpdate

imayhaveborkedit committed:
Rename StateChange enum to Change

imayhaveborkedit committed:
Finish Signal raising conversion

imayhaveborkedit committed:
Add more voice state logging

imayhaveborkedit committed:
Allow for collection of multiple state changes

imayhaveborkedit committed:
Convert remaining emoji to escape sequences

imayhaveborkedit committed:
Whoops

imayhaveborkedit committed:
Fix format

imayhaveborkedit committed:
Another format tweak

imayhaveborkedit committed:
Added FFMPEG log level and log level config option

imayhaveborkedit committed:
Tweaked default invite generation settings

imayhaveborkedit committed:
Added better checks to the bot join announcements

imayhaveborkedit committed:
More improvements to the voice state update class

And an empty voice stateupdate event check

imayhaveborkedit committed:
constructs.py cleanup

imayhaveborkedit committed:
Finally fix VoiceStateUpdate logic

Also added CONNECT and DISCONNECT

imayhaveborkedit committed:
Misc VoiceStateUpdate related cleanup

imayhaveborkedit committed:
Surely this is the last VoiceStateUpdate logic fix

imayhaveborkedit committed:
Use custom LogRecord for shorter relativeCreated

and possible future changes

imayhaveborkedit committed:
Looks like there was still more logic to fix

imayhaveborkedit committed:
This should be the end of the autopause woes

I keep forgetting I sort of shifted the idea of how the VoiceStateUpdate
class works and didn't update the affected code.

imayhaveborkedit committed:
Replace soft restart with hard restart

This fixes the logger duplication issues and and any weird state issues
that could happen

imayhaveborkedit committed:
What the fuck

imayhaveborkedit committed:
Fix some serialization stuff

imayhaveborkedit committed:
More sanity checks organization

imayhaveborkedit committed:
Use the right os.exec* function

imayhaveborkedit committed:
Misc logging tweaks

imayhaveborkedit committed:
Fix windows encoding fuckery

imayhaveborkedit committed:
Fixed failed nested string formatting in cmd_help

imayhaveborkedit committed:
Fix duplicate delete invoking warnings

I think?

imayhaveborkedit committed:
Change diff logging from noise to everything

imayhaveborkedit committed:
Pycharm console colors hack

imayhaveborkedit committed:
Command updates

Catch HTTPException in setname with a notice about the rate limit
Remove owner_only restriction from setnick

imayhaveborkedit committed:
Fix restart function os.exec* again

dan2wik committed:
Fix missing -r in update_dependencies_linux.sh (#595)

ImthePeach committed:
Updates to OS X .command files (#555)

* Update runbot_osx.command

* Update update_dependencies_osx.command

imayhaveborkedit committed:
Add more voicedebug logging and tweak funcs

imayhaveborkedit committed:
Merge branch 'review' of https://github.com/SexualRhinoceros/MusicBot into review

imayhaveborkedit committed:
Improved command parsing...

- Commands are now split by spaces only (previously it was any
whitespace, as per split() default)
- Command functions now accept (*args) and (*, args).  The former will
return the given arguments in a list, similar to how leftover_args
worked.  The latter will rejoin the given arguments back into a string.
- Added _player argument to get non-error raising player (allows None)
- Updated debug command
- Changed some old % formats to .format()

imayhaveborkedit committed:
Fuck this logging info holy shit

imayhaveborkedit committed:
Tweak logging format

imayhaveborkedit committed:
Tweaked format strings and minor cleanup

imayhaveborkedit committed:
Added sanity checks

imayhaveborkedit committed:
More logger changes

Fix logger dupping on restart
Increase relativeCreated precision

imayhaveborkedit committed:
Remove chcp change

This actually does nothing

imayhaveborkedit committed:
Changed Serializable prototype signatures

More generic sigs, no reference to playlist

imayhaveborkedit committed:
Wrote serialization functions for playlist

imayhaveborkedit committed:
Added objgraph debug function

imayhaveborkedit committed:
Remove unneeded exception chain in debug cmd

imayhaveborkedit committed:
Various fixes

Fixed "real" stdout not being set (breaks shutil)
Fixed path being set incorrectly (subprocess is dumb)
Fixed owner permissions not being granted correctly with the auto option
Added additional content-type allowance

imayhaveborkedit committed:
Misc text adjustments

imayhaveborkedit committed:
Added slots and codeblock option to Response

imayhaveborkedit committed:
Changed random module import

from random import ... -> import random

imayhaveborkedit committed:
More voice connection handling

Changed timeouts, added timeout to voice.connect(), reduced tries from
10 to 5, restart if they all failed

imayhaveborkedit committed:
Serialization changes

imayhaveborkedit committed:
Redo serialization

imayhaveborkedit committed:
Added deserialization obj_hook

And change `__name__` to `__qualname__`

imayhaveborkedit committed:
Finish deserialization code

I think... I hope...

imayhaveborkedit committed:
Fixed playlist deserializer

Bother

imayhaveborkedit committed:
Move bin/ appending code to req_ensure_env

imayhaveborkedit committed:
Create queue de/serialization functions

imayhaveborkedit committed:
Added some sanity checks and fixed serialization

Also tweaked status updater

imayhaveborkedit committed:
4Head

imayhaveborkedit committed:
Remove redundant session checks

imayhaveborkedit committed:
Fixed that damn stream entry deserialization bug

Holy shit what an awful ride thanks youtube-dl

imayhaveborkedit committed:
Misc cleanup and message tweaks

imayhaveborkedit committed:
Add queue serialization calls in events

Might still need a periodic serializer but not until progress stuff gets
added (if it gets added)

imayhaveborkedit committed:
Clarify dev command caveat

imayhaveborkedit committed:
Thats a weird bug

imayhaveborkedit committed:
Remove unneeded serialization event

imayhaveborkedit committed:
Fix timedelta garbage

Stops the random high precision outputs

imayhaveborkedit committed:
Change Serializable class arg name

imayhaveborkedit committed:
Add safe_send_message kwargs and handle HTTP err

imayhaveborkedit committed:
Rearrange code and cleanup

And add one serialization field

imayhaveborkedit committed:
Locked version for discord.py

imayhaveborkedit committed:
Moved resuming check into VoiceStateUpdate

imayhaveborkedit committed:
Code cleanup, renames, new function, ...

Cleaned up bits of code
Added some comments and reworded messages
Renamed update_now_playing to update_now_playing_status
Shuffled the autoplaylist in hopes of fixing the rng
Removed unneeded serialization calls involving autoplaylist entries
Added update_now_playing_message function
Sort keys during serialization

imayhaveborkedit committed:
Did I really never notice this

imayhaveborkedit committed:
Changed stream entry data format

This should fix serialization and hopefully not break anything

imayhaveborkedit committed:
Misc code cleanup and changes

Nothing too important
Changed change_status to change_presence
Cleaned up misc code and comments
Fixed some messages and message formatting

imayhaveborkedit committed:
Is this why my voice code was broken?

imayhaveborkedit committed:
Forgot to remove this before

imayhaveborkedit committed:
Update discord.py pinned version

imayhaveborkedit committed:
Update HelpfulError format, added footnote arg

Rip nice formatting

imayhaveborkedit committed:
Fix message typos

imayhaveborkedit committed:
Requests is bad here mmkay

imayhaveborkedit committed:
Added activity indicator during connection phase

imayhaveborkedit committed:
Removed useless ytdl options

imayhaveborkedit committed:
Added bootstrap.py

So far it has the general process laid out, a few steps partially
complete, and stubs for the remaining steps.

imayhaveborkedit committed:
More bootstrap work

Fuyukai committed:
bootstrap.py changes || DO NOT MERGE (#641)

* PEP 8 conformity in bootstrap.py

* Add installation of pip to bootstrap.

* Add basic linux python bootstrapping.

* Reboot into Python 3.5 automatically when it is compiled.

* Terrible attempt at permission stuff

* Download the right target version, and boot into the right executable.

KingGold171 committed:
Feature changes

Added - MaxVolume in permissions (Limit how loud people can set the bots
volume, the number set in the permissions.ini is the limit)

Changed - Volume to display current volume only

Added - SetVolume to set the volume of the bot (works exact same as
volume did.

KingGold171 committed:
Fixed comments

Added maxvolume to the permissions examples comments.

imayhaveborkedit committed:
More bootstrap work

Sort of finished the EnsurePython task and changed some helper stuff

imayhaveborkedit committed:
Added EnsurePython.setup_win32  and

A few other changes

imayhaveborkedit committed:
More work and stubs

I have not tested any of this I hope nothing is horribly broken

imayhaveborkedit committed:
Extra stubs

imayhaveborkedit committed:
Move bootstrap.py

jayktaylor commented on musicbot/bot.py:
```diff
@@ -1464,15 +1464,26 @@ def check(m):
+        if not new_volume:
+            return Response('Current volume: `%s%%`' % int(player.volume * 100), reply=True)
+			
+    async def cmd_setvolume(self, message, player, permissions, new_volume=None):
```
This shouldn't be it's own command. For efficiency, it should be a part of the `volume` command, like it has always been.

KingGold171 replied:
Volume and Setvolume back as one command (MaxVolume permissions still included)

jayktaylor commented on musicbot/bot.py:
```diff
@@ -1482,7 +1493,12 @@ def check(m):
+			
+        if permissions.max_volume and new_volume > permissions.max_volume:
+            raise exceptions.PermissionsError(
+                "[ERROR] You are not able to set the volume above (%s)" % permissions.max_volume
```
Probably remove the `[ERROR]` here, it already says `You do not have permission to use this command.` because of the exception that you're raising.

KingGold171 replied:
What if i was to have the [Error]    is they have permissions to change it but try going above the set limit, and the default if they don't ?

jayktaylor commented on musicbot/permissions.py:
```diff
@@ -98,7 +99,8 @@ def __init__(self, name, section_data):
         self.max_song_length = section_data.get('MaxSongLength', fallback=PermissionsDefaults.MaxSongLength)
         self.max_playlist_length = section_data.get('MaxPlaylistLength', fallback=PermissionsDefaults.MaxPlaylistLength)
-
+        self.max_volume = section_data.get ('MaxVolume', fallback=PermissionsDefaults.MaxVolume)
```
Remove the space after `section_data.get`

KingGold171 replied:
Space removed

KingGold171 committed:
Resolved Requested Changes

+ Volume and SetVolume are back as one command (MaxVolume Permissions
still included)
- space in permissions.py

+- Error message to be worked on

KingGold171 committed:
Atom Changes

Fixes based on Atom's error reporting

"[ERROR]" removed from maxvolume no perms event

KingGold171 committed:
Merge branch 'review' into master
